spec/models/user_spec.rb

```diff
@@ -7,11 +7,30 @@
     expect(create(:user).admin?).to be false
   end
 
-  it 'generates the reset password token' do
-    user = create(:user)
-    expect(Spree::UserMailer).to receive(:reset_password_instructions).with(user, anything, {}).and_return(double(deliver: true))
-    user.send_reset_password_instructions
-    expect(user.reset_password_token).not_to be_nil
+  context "recoverable" do
+    let(:user) { create(:user) }
+
+    it 'generates the reset password token' do
+      expect(Spree::UserMailer).to receive(:reset_password_instructions).with(user, anything, {}).and_return(double(deliver: true))
+      expect { user.send_reset_password_instructions }.to change(user, :reset_password_token).to be_present
+    end
+
+    it "regenerates a spree api key on successful password change" do
+      user.generate_spree_api_key!
+
+      user.password = "123456678"
+      user.password_confirmation = "123456678"
+      expect { user.save! }.to change(user, :spree_api_key).to be_present
```

Comment:
@gmacdougall ah yes good catch. i wanted to have a verification in there that it didn't just clear the api key, but this spec does read oddly seeming to indicate that it is changing it from not present to present. i've made a slight update to be more clear about that